Add AIX 7.2 and rc.d support to ansible

sam-github opened the pull request.

sam-github commented:
This runs to completion, and runs jenkins, but TODO is at least:
- [ ] add new machine to jenkins
- [ ] re-ansible host with the machine secret
- [ ] run builds...  see what remains TBD, which will likely include compiler installs

mhdawson commented on ansible/roles/jenkins-worker/templates/aix.rc2.j2:
```diff
@@ -1,12 +1,34 @@
-export ARCH=ppc64
-export DESTCPU=ppc64
+##################################################
+# name: S20jenkins
```
Maybe delete this line since it is not called S20jenkins anymore?

sam-github commented:
Is our ci behind a firewall?

AIX72 is getting `SEVERE: https://ci.nodejs.org/ provided port:41913 is not reachable`:

```
$ /home/iojs/jdk8u192-b12-jre/bin/java -Xmx128m -Dorg.jenkinsci.plugins.gitcli
ent.Git.timeOut=30 -jar "$HOME/slave.jar" -secret 61acd0525a0490ac3f55fd9bb1e1a885ab13
88347d52d7ec3b8a52c4d0e295a1 -jnlpUrl https://ci.nodejs.org/computer/test-osuosl-aix72
-ppc64_be-1/slave-agent.jnlp
Sep 17, 2019 4:09:43 PM hudson.remoting.jnlp.Main createEngine
INFO: Setting up agent: test-osuosl-aix72-ppc64_be-1
Sep 17, 2019 4:09:43 PM hudson.remoting.jnlp.Main$CuiListener <init>
INFO: Jenkins agent is running in headless mode.
Sep 17, 2019 4:09:43 PM hudson.remoting.Engine startEngine
INFO: Using Remoting version: 3.29
Sep 17, 2019 4:09:43 PM hudson.remoting.Engine startEngine
WARNING: No Working Directory. Using the legacy JAR Cache location: /home/iojs/.jenkins/cache/jars                                                                           
Sep 17, 2019 4:09:43 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Locating server among [https://ci.nodejs.org/]
Sep 17, 2019 4:09:43 PM org.jenkinsci.remoting.engine.JnlpAgentEndpointResolver resolve
INFO: Remoting server accepts the following protocols: [JNLP4-connect, Ping]
Sep 17, 2019 4:09:48 PM org.jenkinsci.remoting.engine.JnlpAgentEndpointResolver isPortVisible                                                                                
WARNING: connect timed out
Sep 17, 2019 4:09:48 PM hudson.remoting.jnlp.Main$CuiListener error
SEVERE: https://ci.nodejs.org/ provided port:41913 is not reachable
java.io.IOException: https://ci.nodejs.org/ provided port:41913 is not reachable
        at org.jenkinsci.remoting.engine.JnlpAgentEndpointResolver.resolve(JnlpAgentEndpointResolver.java:287)                                                               
        at hudson.remoting.Engine.innerRun(Engine.java:523)
        at hudson.remoting.Engine.run(Engine.java:474)
```

I thought the port might be wrong, but checking the jenkins logs for AIX61:
```
INFO: Agent discovery successful
  Agent address: ci.nodejs.org
  Agent port:    41913
...
```

Is there some extra configuration I need to do somewhere, @mhdawson ?

sam-github commented:
@rvagg @mhdawson for about 15 minutes I had a jenkins secret in the log message above. I removed it, but of course its still in the github comment history, and email notfications, etc.

So, I deleted the node to invalidate the secret because I couldn't find a way to change it.

But then, when I created a node again with the same name, it got the same secret. So I deleted the node again.

I'm not sure if there is anything else I should do about the brief secret leak.

I'm also not sure how to create a node with the expected name if the secrets are somehow tied to the node name, though I don't think they should be.

mhdawson commented:
I remember some other time wanting to use a different secret but the end result was that with the same name you get the same secret.  We do have firewall rules which must be enabled by IP to connect as well so simply having the secret does not necessarily allow somebody to connect a machine.  

I think at this point I'd just start the machines at -2 instead of -1 although I'm on the fence if that helps versus just using -1 and counting on the second level of safety provided by the firewall.  The main difference would be is that we might notice -1 if there was not supposed to be one, versus if we use -1 and somebody could replace it.

mhdawson commented:
Yes it is behind a firewall. Info on how is here: https://github.com/nodejs/build/blob/master/jenkins/doc/add-worker-firewall-entry.md.  Let me know if you don't have access to be able to add the rule.

sam-github commented:
@mhdawson I'm not in the infra group, so could you add `140.211.9.30` to the firewall, please?

mhdawson commented:
@sam-github added to firewall.

mhdawson commented:
ran S20jenkins and agent looks connected so firewall entry should be good.

joaocgreis commented on ansible/roles/jenkins-worker/vars/main.yml:
```diff
@@ -5,7 +5,7 @@
 
 init: {
-  aix: 'aix71',
+  aix: ['aix61', 'aix71', 'aix72'],
```
If 6.1 is to remain in `/setup`, doesn't it just add confusion to add it here?

Or, if this can be used for 6.1, delete `/setup/aix61`?

sam-github replied:
I removed 6.1. Should I remove 7.1? We don't have any 7.1 machines, so I doubt it would work (or has even been tested). Mind you, it might work once 7.2 is working.

joaocgreis replied:
If it's reasonable to expect that 7.1 works, possibly with small modifications, I'm ok with this line. Everything in Ansible has to be taken with a grain of salt, we don't have the resources to ensure everything works all the time and this is not our final product in any case.

However if it's more likely that we'll never need this again in the future it's better to remove it, but no point in blocking this PR. So, it's your call!

sam-github commented:
ci: https://ci.nodejs.org/job/node-test-commit-aix72/nodes=aix72-ppc64/3

sam-github commented:
^--- note that https://ci.nodejs.org/job/node-test-commit-aix72 was cloned from base job and reconfigured to only run on aix72, and to clone nodejs/build from https://github.com/sam-github/build/tree/add-aix72

richardlau commented on jenkins/scripts/select-compiler.sh:
```diff
@@ -84,6 +84,12 @@ elif [ "$SELECT_ARCH" = "AIXPPC" ]; then
+	# build tools (for example, python), and for the compiled binaries
+	# (node and intermediate tools). Note that 8.x will be EOL at end of
+	# 2019, so there is little reason to get this working, and it should be
+	# removed in 2020.
```
Are there any plans to run any of the 8.x jobs on AIX 7.2? It might be easier/simpler to keep those on the 6.1 boxes if it's only until the end of the year.

sam-github replied:
no, but I thought the text might be useful if anybody ever did try. removed it.

sam-github commented:
A @mhdawson can you put these build artifacts from `test-osuosl-aix72-ppc64_be-2:/opt` in  https://ci.nodejs.org/downloads/aix72:
1. gcc-6.3-aix7.2.ppc.tar.gz
2. ccache-3.7.4.aix7.2.ppc.tar.gz

richardlau commented on jenkins/scripts/select-compiler.sh:
```diff
@@ -73,8 +73,24 @@ elif [ "$SELECT_ARCH" = "S390X" ]; then
+        export PATH="/opt/ccache-3.7.4/libexec:/opt/gcc-6.3/bin:$PATH"
+        export CC="`which gcc`" CXX="`which g++`" CXX_host="`which g++`"
+        echo "Compiler set to 6.3"
+	return
```
Indentation is off here.

richardlau commented on jenkins/scripts/select-compiler.sh:
```diff
@@ -73,8 +73,24 @@ elif [ "$SELECT_ARCH" = "S390X" ]; then
+	return
+      else
+        echo "Compiler defaulting to 4.8"
+	return
```
Indentation is off here.

richardlau commented on jenkins/scripts/select-compiler.sh:
```diff
@@ -73,8 +73,24 @@ elif [ "$SELECT_ARCH" = "S390X" ]; then
+        echo "Compiler set to 6.3"
+	return
+      else
+        echo "Compiler defaulting to 4.8"
```
This isn't correct on AIX 7.2? The compiler installed via yum from the AIX Toolbox on AIX 7.2 is gcc 8.

sam-github replied:
fixed

sam-github commented:
We can't start using aix7.2 in node builds until we have a couple more machines. The setup of those machines will also help figure out how reproducible the install process is.

I've a CI running now: https://ci.nodejs.org/job/node-test-commit-aix72/5/nodes=aix72-ppc64/

@mhdawson @richardlau
